compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -2008,6 +2020,79 @@ public void visit(BLangXMLNavigationAccess xmlNavigation) {
         }
     }
 
+    @Override
+    public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
+        analyzeInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList, env);
+    }
+
+    private void analyzeInterpolations(List<BLangExpression> sequenceList, SymbolEnv env) {
```

Comment:
Can you check whether we can have common logic for IsolationAnalyzer, DataflowAnalyzer, CodeAnalyzer and TypeChecker; maybe extract out the atom lists and then visit? wdyt?